### ISSUE TITLE
Rule Closing brace indented incorrectly fixed automatically incorrectly

### ISSUE THREAD

**Issue by vitalyzhakov** [squizlabs/PHP_CodeSniffer#2447]
Hello!

I have intresting case (example)

```php
<?php


?>


                        <div class="top-nav__hide-column top-nav__hide-column-close top-nav__hide-column-<?php echo $key + 1; ?> top-nav__hide-column_fixed-width <?php if ($key + 1 == count($menulink->children->imagemenu)) : ?> top-nav__hide-column_removed <?php endif; ?>">
```
phpCs found an error:
 9 | ERROR   | [x] Closing brace indented incorrectly; expected 23 spaces,
   |         |     found 40

and fix it to
```php
<?php


?>


                        <div class="top-nav__hide-column top-nav__hide-column-close top-nav__hide-column-<?php echo $key + 1; ?> top-nav__hide-column_fixed-width <?php if ($key + 1 == count($menulink->children->imagemenu)) : ?> top-nav__hide-column_removed <?php 
                       endif; ?>">
```

I think, rule works incorrectly.

**Comment by xuanquynh** [comment]
Is it necessary to solve this case? I think no developer writes this funny code and use phpcbf to fix. :)

**Comment by gsherwood** [comment]
I think the error is actually "Closing brace must be on a line by itself", which is why the fix is to put the closing brace (the `endif` in this case) on a line by itself. It's also valid, so the sniff is working correctly.

I would suggest not checking template files like this using the same standards you use for PHP class files. They are almost always incompatible.

But if you want to, I'd suggest assigning values to variables and then using those variables inside your HTML. Something like this maybe:
```php
<?php
$newKey = ($key + 1);
$extraClasses = '';
if ($newKey == count($menulink->children->imagemenu)) {
    $extraClasses = ' top-nav__hide-column_removed';
}
?>
<div class="top-nav__hide-column top-nav__hide-column-close top-nav__hide-column-<?= $newKey; ?> top-nav__hide-column_fixed-width <?= $extraClasses; ?>">
```
